test/mocks/data/firestore_post.dart

```diff
@@ -68,4 +77,52 @@ class FirestorePostGenerator {
       ),
     );
   }
+
+  /// Add [n] posts at position [location] and return their data and the [PostFirestore] objects
+  Future<List<PostFirestore>> addPosts(
+    FirebaseFirestore firestore,
+    GeoPoint location,
+    int n,
+  ) async {
+    final List<PostFirestore> fakePosts = generatePostsAt(location, n);
+    await setPostsFirestore(fakePosts, firestore);
+    return fakePosts;
+  }
+
+  /// Add [n] posts at position [location] and return their data
+  Future<List<PostData>> addPostsReturnDataOnly(
+    FirebaseFirestore firestore,
+    GeoPoint location,
+    int n,
+  ) async {
+    return (await addPosts(firestore, location, n)).map((e) => e.data).toList();
```

Comment:
I think this can have a slightly improved readability:
```suggestion
    final posts = await addPosts(firestore, location, n);
    return posts.map((post) => post.data).toList();
```